android/src/main/java/com/github/wuxudong/rncharts/markers/RNConditionalMarkerImage.java

```diff
@@ -0,0 +1,127 @@
+package com.github.wuxudong.rncharts.markers;
+
+import android.content.Context;
+import android.graphics.drawable.Drawable;
+import android.widget.RelativeLayout;
+import android.widget.TextView;
+import android.graphics.Canvas;
+
+import com.facebook.react.bridge.ReadableMap;
+import com.facebook.react.bridge.ReadableArray;
+import com.github.mikephil.charting.charts.Chart;
+import com.github.mikephil.charting.components.MarkerImage;
+import com.github.mikephil.charting.data.CandleEntry;
+import com.github.mikephil.charting.data.Entry;
+import com.github.mikephil.charting.highlight.Highlight;
+import com.github.mikephil.charting.utils.MPPointF;
+import com.github.mikephil.charting.utils.Utils;
+import com.github.wuxudong.rncharts.R;
+import com.github.wuxudong.rncharts.highlight.HighlightWithMeta;
+import com.github.wuxudong.rncharts.utils.ConversionUtil;
+
+import java.util.List;
+import java.util.Set;
+import java.util.Map;
+import java.util.HashMap;
+import java.util.HashSet;
+
+/**
+ * Allows skipping drawing of image marker based on conditions.
+ * 
+ * Image marker is skipped for an entry with highlight if any of these is true:
+ *  1. highlight is on a excluded dataset
+ *  2. entry's markerHighlightSource does not match highligh's source
+ */
+public class RNConditionalMarkerImage extends MarkerImage {
+    
+    private boolean mSkipDraw;
+    /*
+     * Map of dataIndex to a set of dataSetIndex of that data, representing 
+     * the datasets to be excluded for showing marker. 
+     *
+     * Null set indicates all dataset.
+     */
+    private Map<Integer, Set<Integer>> mExcludes;
+    
+    private Map<Integer, Set<Integer>> parseExcludesProp(ReadableArray excludesProp) {
+        if (excludesProp != null) {
+            List excludeList = ConversionUtil.toList(excludesProp);
+            Map<Integer, Set<Integer>> excludes =  new HashMap<Integer, Set<Integer>>();
+
+            for (Object exclude : excludeList) {
+                Map excludeMap = (Map)exclude;
+                List dataSetIndexesArray = (List)excludeMap.get("dataSetIndexes");
```

Comment:
how does `excludeMap` work with both `dataSetIndexes` and `dataIndex`? If my data looks ...
```
{:dataIndex0 {:dataSets [A B]}
 :dataIndex1 {:dataSets [C D]}
```
and want to exclude A and D, then it seems that the current implementation wouldn't support it?

Reply:
you would pass in a excludes that looks like:
```
[{:dataIndex 0 :dataSetIndexes [A]}
 {:dataIndex 1 :dataSetIndexes [D]}
```
Would that work?

Reply:
ahh, yes, sorry! Misread, and didn't realize we were passing through the array of maps! thanks!